Saule/Queries/Pagination/PaginationQuery.cs

```diff
@@ -35,7 +35,9 @@ public PaginationQuery(PaginationContext context, object value)
             {
                 // we also should add firstPage as if it's not a zero, then total page count should be shifted based on firstPageNumber
                 // with 60 elements and 20 page size. if firstPage is 0, then lastPage should be 2. If firstPage is 1, then lastPage should be 3
-                var totalPages = (int)Math.Ceiling((double)context.TotalResultsCount / context.PerPage.Value) - 1 + context.FirstPageNumber;
+                int totalPages = context.TotalResultsCount == 0
```

Comment:
What about `context.TotalResultsCount <= context.PerPage.Value`?

Reply:
It should follow existing logic and totalPages would be `1`. So `0` is just edge case that needed to be handled